src/lotus/message/ipc.rs

```diff
@@ -36,21 +39,28 @@ pub struct IPCReadSubnetActorStateResponse {
 }
 
 /// SubnetInfo is an auxiliary struct that collects relevant information about the state of a subnet
+///
+/// Note that the serialization and deserialization casing are different. Reason because for deserialization,
+/// it is from the fvm actor, which is `PascalCase`. When serialize, we are using rust's default casing
 #[derive(Debug, Serialize, Deserialize)]
-#[serde(rename_all = "PascalCase")]
 pub struct SubnetInfo {
     /// Id of the subnet.
+    #[serde(rename(deserialize = "ID"))]
     #[serde(deserialize_with = "deserialize_subnet_id_from_map")]
-    #[serde(rename = "ID")]
+    #[serde(serialize_with = "serialize_subnet_id_to_str")]
     pub id: SubnetID,
     /// Collateral staked in the subnet.
+    #[serde(rename(deserialize = "Stake"))]
     #[serde(deserialize_with = "deserialize_token_amount_from_str")]
+    #[serde(serialize_with = "serialize_token_amount_to_atto")]
     pub stake: TokenAmount,
     /// Circulating supply available in the subnet.
+    #[serde(rename(deserialize = "CircSupply"))]
     #[serde(deserialize_with = "deserialize_token_amount_from_str")]
-    #[serde(rename = "CircSupply")]
+    #[serde(serialize_with = "serialize_token_amount_to_atto")]
```

Comment:
Why do we need this additional deserialization step? Can't we deserialize directly into token amount (and change SubnetInfo if needed)

Reply:
TokenAmount does not implement `Serialize`, without the `serialize_with`, it's serialized into byte array which is definitely hard to read.